File: model-client/src/main/java/org/modelix/model/client/IIndirectBranch.java
package org.modelix.model.client;

import org.modelix.model.api.IBranch;
import org.modelix.model.api.IBranchListener;

public interface IIndirectBranch {
  IBranch getBranch();
  void addListener(IBranchListener l);
  void removeListener(IBranchListener l);
}

File: model-client/src/main/java/org/modelix/model/client/IIndirectBranch.kt
package org.modelix.model.client

import org.modelix.model.api.IBranch
import org.modelix.model.api.IBranchListener

interface IIndirectBranch {
    val branch: IBranch?
    fun addListener(l: IBranchListener?)
    fun removeListener(l: IBranchListener?)
}
